frontend/src/Onboarding/GenericOnboarding.tsx

```diff
@@ -151,6 +158,25 @@ const SelectableCourse: React.FC<SelectableCourseProps> = ({
   );
 };
 
+const ControlledSelectableCourse: React.FC<SelectableCourseProps & {
```

Comment:
The above `SelectableCourse` tries to be an [uncontrolled component](https://reactjs.org/docs/uncontrolled-components.html), but according to the MUI docs, [controlled components take an `onChange`](https://mui.com/components/checkboxes/#controlled).

I had tried using `SelectableCourse` as it had been used before, but for whatever reason I couldn't get it to be uncontrolled (clicking checkbox would not check the component). I'm guessing it's because we pass it an `onChange`? Though I'm not sure, since that seems to be elsewhere... it may also be context dependent (i'm using it in a function component, the others are classes).

TLDR was having trouble getting uncontrolled to work, so I made a controlled variant of `SelectableCourse`. 

If reviewers think it's cleaner, I can also add a `controlled` prop to the existing `SelectableCourse` component so that there's less code duplication.

Reply:
interesting, why do you think we are doing uncontrolled select components for some areas then(completed course screen for example)? if react suggests controlled components what makes our case unique in requiring an uncontrolled select component?

Reply:
What do you think about abstracting the component into a general select with the `isControlled` prop but also exporting `UncontrolledSelectableCourse` and `ControlledSelectableCourse` that essentially just wrap around the generic select and pass down true or false for the `isControlled` prop. 

If you find that clean that would be awesome! Otherwise just adding the prop is fine but lets maybe abstract out the common MUI styling and stuff :)

Reply:
not sure. my best guess is that at the time, it was easier to do, or that we weren't aware of doing so.

and yep! i may also just replace all the instances of the uncontrolled variant, if it's easy enough to do.

Reply:
sweet thanks!!